Mathlib/RingTheory/SimpleRing/Basic.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.SimpleRing.Defs
+import Mathlib.Algebra.Field.Equiv
+import Mathlib.Algebra.Ring.Subring.Basic
+
+/-! # Basic Properties of Simple rings
+
+A ring `R` is **simple** if it has only two two-sided ideals, namely `0` and `⟨1⟩`.
+
+## Main results
+
+- `IsSimpleRing.nontrivial`: simple rings are non-trivial.
+- `IsSimpleRing.of_divisionRing`: division rings are simple.
+- `IsSimpleRing.center_isField`: the center of a simple ring is a field.
+
+-/
+
+variable (R : Type*) [NonUnitalNonAssocRing R]
+
+namespace IsSimpleRing
+
+variable {R}
+
+instance [IsSimpleRing R] : IsSimpleOrder (TwoSidedIdeal R) := IsSimpleRing.simple
+
+instance nontrivial [simple : IsSimpleRing R] : Nontrivial R := by
+  obtain ⟨x, hx⟩ := SetLike.exists_of_lt (bot_lt_top : (⊥ : TwoSidedIdeal R) < ⊤)
+  have h (hx : x = 0) : False := by simp_all [TwoSidedIdeal.zero_mem]
+  use x, 0, h
+
+lemma one_mem_of_ne_bot {A : Type*} [NonAssocRing A] [IsSimpleRing A] (I : TwoSidedIdeal A)
+    (hI : I ≠ ⊥) : (1 : A) ∈ I :=
+  (eq_bot_or_eq_top I).resolve_left hI ▸ ⟨⟩
+
+lemma one_mem_of_ne_zero_mem {A : Type*} [NonAssocRing A] [IsSimpleRing A] (I : TwoSidedIdeal A)
+    {x : A} (hx : x ≠ 0) (hxI : x ∈ I) : (1 : A) ∈ I :=
+  one_mem_of_ne_bot I (by rintro rfl; exact hx hxI)
+
+lemma of_eqBotOrEqTop [Nontrivial R] (h : ∀ I : TwoSidedIdeal R, I = ⊥ ∨ I = ⊤) :
+    IsSimpleRing R where
+  simple := { eq_bot_or_eq_top := h }
+
+instance _root_.DivisionRing.isSimpleRing (A : Type*) [DivisionRing A] : IsSimpleRing A :=
+  .of_eqBotOrEqTop <| fun I ↦ by
+    rw [or_iff_not_imp_left]
+    intro H
+    suffices mem : 1 ∈ I from eq_top_iff.2 <| fun x _ ↦ mul_one x ▸ I.mul_mem_left x _ mem
```

Comment:
The bit after `<|` should be factored out as a (collection of) lemmas. That is, we should have:
```lean
namespace TwoSidedIdeal

lemma one_mem_iff : 1 ∈ I ↔ I = ⊤ := sorry
alias ⟨eq_top, one_mem⟩ := one_mem_iff
```

Reply:
done